### PR TITLE
boards/adafruit-itsybitsy-m4: Model kconfig

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

This models Kconfig for the samd5x based board adafruit-itsybitsy-m4.

It also adds `boards_common_samdx1-arduino-bootloader` and `usb_board_reset`.

### Testing procedure

Check the Kconfig changes and modelling
Green CI

### Issues/PRs references

Part of #16875

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Using murdock to check all apps for the board.

File: boards/common/samdx1-arduino-bootloader/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config MODULE_BOARDS_COMMON_SAMDX1-ARDUINO-BOOTLOADER
+    bool
+    depends on TEST_KCONFIG
+    help
+       Common code of NRF51 boards
```
Comment (benpicco) [inline boards/common/samdx1-arduino-bootloader/Kconfig:11]: SAM D boards

File: boards/adafruit-itsybitsy-m4/Kconfig [hunk 2]
```diff
@@ -22,7 +22,12 @@ config BOARD_ADAFRUIT_ITSYBITSY_M4
     select HAS_PERIPH_TIMER
     select HAS_PERIPH_UART
     select HAS_PERIPH_USBDEV
+    select HAS_HIGHLEVEL_STDIO
```
Comment (leandrolanzieri) [inline boards/adafruit-itsybitsy-m4/Kconfig:25]: This is already there

File: boards/adafruit-itsybitsy-m4/Kconfig [hunk 3]
```diff
@@ -22,7 +22,12 @@ config BOARD_ADAFRUIT_ITSYBITSY_M4
...
+    select MODULE_BOARDS_COMMON_SAMDX1-ARDUINO-BOOTLOADER if MODULE_STDIO_CDC_ACM
```
Comment (leandrolanzieri) [inline boards/adafruit-itsybitsy-m4/Kconfig:26]: Probably we could instead:
```
config MODULE_BOARDS_COMMON_SAMDX1-ARDUINO-BOOTLOADER
   [...]
   default y if MODULE_STDIO_CDC_ACM
```
in `samdx1-arduino-bootloader/Kconfig`

File: sys/usb_board_reset/Kconfig [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 HAW Hamburg
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply sys/usb_board_reset/Kconfig:1]: Done

File: boards/adafruit-itsybitsy-m4/Kconfig (hunk 2 above)
Comment (MrKevinWeiss) [reply boards/adafruit-itsybitsy-m4/Kconfig]: removed

File: boards/adafruit-itsybitsy-m4/Kconfig (hunk 3 above)
Comment (MrKevinWeiss) [reply boards/adafruit-itsybitsy-m4/Kconfig]: changed

File: boards/common/samdx1-arduino-bootloader/Kconfig (hunk 1 above)
Comment (MrKevinWeiss) [reply boards/common/samdx1-arduino-bootloader/Kconfig]: fixed

**Comment by MrKevinWeiss** [comment]
Lets see what murdock says

**Comment by leandrolanzieri** [comment]
> Lets see what murdock says

Looks like it agrees. Please squash @MrKevinWeiss

**Comment by MrKevinWeiss** [comment]
Hmmm looks like I did (auto-merge? :smile: )

**Review by leandrolanzieri — APPROVED** [review]
ACK

**Comment by MrKevinWeiss** [comment]
Thanks for the review!

**Pull request merged**